composer/trainer/dist_strategy.py

```diff
@@ -204,13 +204,6 @@ def prepare_fsdp_module(model: torch.nn.Module, optimizers: Optional[Union[torch
     else:
         raise ValueError(f'Unable to interpret mixed_precision={mixed_precision}')
 
-    if sharding_map_key != 'NO_SHARD' and (precision == Precision.AMP_FP16 and param_dtype != torch.float16 or
-                                           precision == Precision.AMP_BF16 and param_dtype != torch.bfloat16):
-        raise ValueError(
-            f'FSDP in PyTorch 1.13 does not support precision `{precision}` with sharding strategy `{sharding_strategy}` '
-            f'and param_dtype `{param_dtype}.` param_dtype needs to match the precision data type `{precision}.` '
-            "Please adjust the MixedPrecision parameter, for example `mixed_precision='PURE'`.")
-
```

Comment:
ppl can still pass mixed_precision as a dict and to shoot themselves in the foot. 
If you still want to remove the error, maybe convert it into warning instead

Reply:
otherwise torch will spit out that massive non-sensical stack trace.